### ISSUE TITLE
Possible bug on liveness check

### ISSUE THREAD

**Issue by italovalcy** [kytos-ng/of_lldp#106]
Hi,

Using Kytos-ng 2022.3 we noticed the following situation on the logs:

```
2024-10-12 01:07:02,550 - INFO [kytos.napps.kytos/topology] [main.py:579:handle_link_liveness_status] (thread_pool_app_1018) Link liveness up: Link(Interface('novi_port_6', 6, Switch('00:00:00:00:00:28:00:01')), Interface('novi_port_16', 16, Switch('00:00:00:00:00:00:00:10')))
2024-10-12 01:07:02,553 - INFO [kytos.napps.kytos/topology] [main.py:579:handle_link_liveness_status] (thread_pool_app_37) Link liveness down: Link(Interface('novi_port_6', 6, Switch('00:00:00:00:00:28:00:01')), Interface('novi_port_16', 16, Switch('00:00:00:00:00:00:00:10')))
2024-10-12 01:07:06,562 - INFO [kytos.napps.kytos/mef_eline] [main.py:674:handle_link_down] (thread_pool_app_798) Event handle_link_down Link(Interface('novi_port_16', 16, Switch('00:00:00:00:00:00:00:10')), Interface('novi_port_6', 6, Switch('00:00:00:00:00:28:00:01')))
```

Two points to notice on the logs:
- The liveness link UP event was generated before the Down event.. which seems to be is a bug, not sure the conditions that led to this bug
- We have a pretty flexible liveness check configuration in place (LIVENESS_DEAD_MULTIPLIER = 20  and default pooling interval POLLING_TIME = 3) so not sure if this was a real link/forwarding failure or not, but most likely not a forwarding issue because it very weird that the link stayed for 60s down and returned exactly on the moment it was considered down

The impacts of this was that EVCs having that link as the primary path stayed down ever since the event happened. Additionally, EVCs with backup path never come back to the primary path after this failure.

**Comment by viniarck** [comment]
Thanks for catching this, @italovalcy. I'll take a look on this one later today. Indeed with this polling time and dead multiplier, it shouldn't not be down after 3 secs. 

1 - Do you also have other logs of event down?
2 - Can you elaborate on a bit more how the link down happened? I'm not sure if I followed what you meant

> but most likely not a forwarding issue because it very weird that the link stayed for 60s down and returned exactly on the moment it was considered down

In the meantime I'll see if I can stop races conds too.

**Comment by italovalcy** [comment]
> Thanks for catching this, @italovalcy. I'll take a look on this one later today. Indeed with this polling time and dead multiplier, it shouldn't not be down after 3 secs.
> 
> 1 - Do you also have other logs of event down? 2 - Can you elaborate on a bit more how the link down happened? I'm not sure if I followed what you meant
> 
> > but most likely not a forwarding issue because it very weird that the link stayed for 60s down and returned exactly on the moment it was considered down
> 
> In the meantime I'll see if I can stop races conds too.

Hi Vinicius! thanks for looking into this.

1 - no other event of link down. this scenario is configured over another L2VPN from the carrier provider, so no physical link down event was actually expected.

2 - what I meat was basically: since our POOLING_INTERVAL=3s and LIVENESS_DEAD_MULTIPLIER=20, it means the link have to stay down for 60s before triggering the liveness link down event, right? if that is true, I think it very unlikely that the link stayed for 60s actually not forwarding traffic and then returned to UP right after sending the link liveness down event.

**Comment by viniarck** [comment]
> it means the link have to stay down for 60s before triggering the liveness link down event, right?

Yes, if `pooling * dead_multiplier` passes since the last hello received it consider liveness down. You're correct, if it just went up, received a hello, this down shouldn't have happened. I'll investigate. They're a running on diff threads in the example too, maybe it's a race cond. We'll see.
